### PR TITLE
boards: arm: bl654_usb: Fix USB enablement issues

### LINKED ISSUE #39904: bl654_usb does not work with hci_usb sample application
**Describe the bug**
As noted in https://github.com/zephyrproject-rtos/zephyr/issues/38245 the hci_uart sample application does not work when building for bl654_usb due to the USB stack being used for console output

**Expected behavior**
For the sample application to work

### PR DESCRIPTION
**Pull request by nordicjm**
This addresses USB issues with the BL654 USB dongle which prevent USB
from working properly (e.g. hci_usb) or do not automatically give
access to the console of the board (most other samples) by adding a
feature that will automatically enable the USB peripheral on startup
without needing to be manually added to sample/user applications.

Signed-off-by: Jamie McCrae <jamie.mccrae@lairdconnect.com>

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/39904 and addresses https://github.com/zephyrproject-rtos/zephyr/issues/39969

### CODE REVIEW

File: boards/arm/bl654_usb/Kconfig.defconfig [hunk 1]
```diff
@@ -26,7 +26,10 @@ config FLASH_LOAD_OFFSET
 if USB_DEVICE_STACK
 
 config USB_UART_CONSOLE
-	default y
+	default y if !USB_DEVICE_BLUETOOTH
```
Comment (jfischer-no) [inline boards/arm/bl654_usb/Kconfig.defconfig:29]: Changes in this file should be in own commit, which fixes #39904.

**Comment by jfischer-no** [comment]
Apart from https://github.com/zephyrproject-rtos/zephyr/pull/39987#discussion_r743492792, this patch is a selfish approach, which will also be a bad example for other boards. I am looking at how we can make it better and bring USB device support initialization during startup back in case only CDC ACM class is enabled and CDC ACM UART is used as logging/shell backend.

File: boards/arm/bl654_usb/Kconfig.defconfig (hunk 1 above)
Comment (carlescufi) [reply boards/arm/bl654_usb/Kconfig.defconfig:29]: @lairdjm could you please do that as suggested by @jfischer-no?

Comment (lairdjm) [reply boards/arm/bl654_usb/Kconfig.defconfig:29]: @carlescufi this is awaiting a USB fix from @jfischer-no

Comment (jfischer-no) [reply boards/arm/bl654_usb/Kconfig.defconfig:29]: For completeness, option USB_UART_CONSOLE is removed. Shield approach in #40645 was not accepted. I opened #40645 to continue the discussion.

Anyway, I do not want to accept any USB device support setting at board level anymore. The boards in Zephyr tree should be in minimal configuration, https://docs.zephyrproject.org/latest/guides/porting/board_porting.html#general-recommendations, so no configuration to enable USB support or USB device classes.

**Comment by lairdjm** [comment]
@carlescufi can we progress with this temporary fix for the upcoming release until there is a proper global fix for this? As this basically means the bl654_usb board is dead without manual changes to customer codebases

**Comment by lairdjm** [comment]
Fixed in https://github.com/zephyrproject-rtos/zephyr/pull/42371
